integration/types/datatype_test.rb

```diff
@@ -497,11 +494,9 @@ def test_can_insert_tuple_type_all_collection_datatypes_nil
 
       # Create the input
       params = [0]
-      DatatypeUtils.collection_types.each do |collection_type|
-        DatatypeUtils.primitive_datatypes.each do |datatype|
-          unless collection_type == 'Tuple'
-            params.push(nil)
-          end
+      DatatypeUtils.collection_types.each do
```

Comment:
I haven't been paying very close attention, but it seems to me that there's a lot of duplicated logic throughout this file. If you consolidated some of it to methods, then in the future, changes like special-casing 'Tuple' (or vice versa) would only require changing one place.

Mull it over and do what you think is best.